us-autocomplete-api/client.go

```diff
@@ -0,0 +1,50 @@
+package autocomplete
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/smartystreets/smartystreets-go-sdk"
+)
+
+// Client is responsible for sending batches of addresses to the us-street-api.
+type Client struct {
+	sender sdk.RequestSender
+}
+
+// NewClient creates a client with the provided sender.
+func NewClient(sender sdk.RequestSender) *Client {
+	return &Client{sender: sender}
+}
+
+// SendBatch sends the batch of inputs, populating the output for each input if the batch was successful.
+func (c *Client) SendLookup(lookup *Lookup) error {
+	if lookup == nil || len(lookup.Prefix) == 0 {
+		return nil
+	} else if response, err := c.sender.Send(buildRequest(lookup)); err != nil {
+		return err
+	} else {
+		return deserializeResponse(response, lookup)
+	}
+}
+
+func deserializeResponse(response []byte, lookup *Lookup) error {
```

Comment:
Great question! There definitely could be a separate serialization implementation, which would allow plug-n-play serialization for those that want to use a different json package. That's probably a different project though (as it would be used by all 3 API clients).